lib/cc/cli/analyze.rb

```diff
@@ -20,23 +20,26 @@ def initialize(_args = [])
         @path_options = []
 
         process_args
-        apply_config_options
       end
 
       def run
-        require_codeclimate_yml
+        formatter.started
 
         Dir.chdir(MountedPath.code.container_path) do
-          runner = EnginesRunner.new(registry, formatter, source_dir, config, path_options)
-          runner.run
+          engines.each do |engine|
+            formatter.engine_running(engine) do
+              run_engine(engine)
+            end
+          end
         end
-      rescue EnginesRunner::NoEnabledEngines
-        fatal("No enabled engines. Add some to your .codeclimate.yml file!")
+
+        formatter.finished
+      ensure
+        formatter.close if formatter.respond_to?(:close)
```

Comment:
this could also just be defined as a no-op for the formatters that don't need to close, if it'd be better to have a uniform "formatter" interface.